Automattic/pocket-casts-android
# [Project] Manage Downloaded Episodes - Display Modal when running on low storage and opening Storage & Data Use screen

## Description
- Displays the low storage modal when running on low storage and opening the **Storage & Data Use** screen
- This PR does not implement the snooze for 7 days feature when taping on `Maybe Later` button
- See: VvERCFIopLjFL8Fnczsf8r-fi-123_1167
- Context: p1730298732804909/1730291928.354409-slack-C07TL9YK9V1

Fixes #3098

## Testing Instructions
Modify the code bellow to always return `true` to make the tests faster. This will mock the low storage check
https://github.com/Automattic/pocket-casts-android/blob/565d4f7ef92be14a27bc7cfe3c6aa6a24962a178/modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/StorageUtil.kt#L11

1. Download an episode
2. Go to Profile -> Storage & Data Use
3. ✅ Ensure the modal opens 
4. Tap on Maybe Later
5. ✅ Ensure the modal dismiss
6. Tap on Manage Downloads
7. ✅ Ensure the Downloaded files screen opens 

## Screenshots or Screencast 
[Screen_recording_20241101_111721.webm](https://github.com/user-attachments/assets/7bfe0fb6-4f31-4e5b-9bbf-5c121cf06e69)


| | | |
|--------|--------|--------|
| <img src="https://github.com/user-attachments/assets/1e530f4b-c0e9-4c40-b1e2-cd6ae1f76697" width="400" alt="Screenshot_20241101_111138"/> | <img src="https://github.com/user-attachments/assets/92b9fc5e-d090-4af9-9293-54ca51ee9f40" width="400" alt="Screenshot_20241101_111230"/> | <img src="https://github.com/user-attachments/assets/e20ed83c-9f05-4088-ad5b-8406c790b8dc" width="400" alt="Screenshot_20241101_111253"/> | 


| | | |
|--------|--------|--------|
| <img src="https://github.com/user-attachments/assets/3b1bd42f-e274-4de9-8f41-7c71d7738ed7" width="400" alt="light"/> | <img src="https://github.com/user-attachments/assets/9ab8e51a-21d6-486a-b76f-677827c06897" width="400" alt="Screenshot_20241101_111057"/> | <img src="https://github.com/user-attachments/assets/a1ed2d0d-1fec-4377-ae60-89bdccb1c162" width="400" alt="Screenshot_20241101_111115"/> | 






## Checklist
 
#### I have tested any UI changes...
